python/opae.admin/opae/admin/tools/regmap_debugfs.py

```diff
@@ -0,0 +1,166 @@
+CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+"""
+
+import argparse
+import os.path
+
+
+class RegmapDebugfs():
+    """RegmapDebugfs is a class for accessing the debugfs
+       inteface of regmaps in the linux kernel"""
+
+    def __init__(self, regmapdir):
+
+        self.registers_path = os.path.join(regmapdir, 'registers')
+        self.range_path = os.path.join(regmapdir, 'range')
+        self.ranges = []
+        self.record_len = 0
+        self.register_stride = 0
+
+    def get_register_info(self):
+        """get information about regmap"""
+        with open(self.registers_path, 'r') as _file:
+            line0 = _file.readline()
+            line1 = _file.readline()
+            self.record_len = len(line0)
+            list0 = line0.strip().split(': ')
+            list1 = line1.strip().split(': ')
+            self.register_stride = int(list1[0], 16) - int(list0[0], 16)
+
+    def get_range_info(self):
+        """get the list of address ranges for the regmap"""
+        with open(self.range_path, 'r') as _file:
+            for line in _file.readlines():
+                list0 = line.strip().split('-')
+                range_dict = {
+                    'min': int(list0[0], 16),
+                    'max': int(list0[1], 16),
+                    'num_records': ((int(list0[1], 16) - int(list0[0], 16)) /
+                                    self.register_stride) + 1,
+                }
+                self.ranges.append(range_dict)
+
+    def reg_read(self, addr):
+        """reg_read returns the value read from the given
+           address in the regmap"""
+
+        found = False
+        num_records = 0
+        for range_dict in self.ranges:
+            if range_dict['min'] <= addr <= range_dict['max']:
+                found = True
+                offset = (addr - range_dict['min'])
+                num_records = num_records + offset / self.register_stride
+                break
+            else:
+                num_records = num_records + range_dict['num_records']
+
+        if not found:
+            raise Exception("bad address")
+
+        with open(self.registers_path, 'r') as _file:
+            _file.seek(num_records * self.record_len)
+            line0 = _file.readline()
+
+        list0 = line0.strip().split(': ')
+        raddr = int(list0[0], 16)
+
+        if raddr != addr:
+            msg = "bad addr read back 0x{:x} != 0x{:x}".format(raddr, addr)
+            raise Exception(msg)
+
+        return int(list0[1], 16)
+
+    def reg_write(self, addr, val):
+        """reg_write writes val to the addr in the regmap"""
+
+        found = False
```

Comment:
This one can be shortened to something similar to:
```
for range_dict in self.ranges:
    if range_dict['min'] <= addr <= range_dict['max']:
        break
else:
    raise Exception("bad address")
```
This is looping over all addresses. If none of addresses causes break of loop, it will execute `else` clause.

Reply:
This looks cool, but I'll use sys.exit.